sklearn/utils/_array_api.py

```diff
@@ -10,6 +10,10 @@
 from .fixes import parse_version
 
 
+def XP_FLOAT_DTYPES(xp):
+    return xp.float64, xp.float32, xp.float16
```

Comment:
Should we include float16 for all backends?

Reply:
There is already `sklearn._array_api.supported_float_dtypes(xp)` that includes `float16` only when supported by the backend. I think the new `XP_FLOAT_DTYPES` is redundant.

Reply:
float16 is supported by torch, but torch.sqrt is not supported for float16. How should we handle this?

Reply:
OK, when mean and var are computed with float64, torch.sqrt should also work.